### ISSUE TITLE
windows 下jd-gui貌似默认是gbk编码

### ISSUE THREAD

**Issue by mousycoder** [java-decompiler/jd-gui#129]
我打得jar包是utf-8，用其他软件打开里面的xml中文注释是正常的，但是用windows下的jd-gui是乱码，不知道jd-gui怎么设置编码

**Comment by donizyo** [comment]
你用中文问问题怎么会有人回答你呀，傻孩子？

**Comment by samrocketman** [comment]
Looks like the OP (used google translate) means:
> JD-GUI appears to open their Jar file in GBK encoding.  However, their xml files were encoded with UTF-8 so the characters don't look right.

**Comment by donizyo** [comment]
Encoding problem maybe related to `getBytes(Charset)`

**Comment by zhulong1023** [comment]
wait for a reply

**Comment by mqiezi** [comment]
我觉得这是培训老外中文的好机会

**Comment by liach** [comment]
呃，这个坑似乎已经没人填了

**Comment by donizyo** [comment]
@mqiezi 可以的，出个教中文的人工智能吧

**Comment by ice1000** [comment]
I don't think it good to use Chinese on GitHub, especially when you're issuing a non-Chinese project.

**Comment by JustGregory** [comment]
@ice1000 No, this only means there is now a market for further I18N/L10N into more languages.... ;-p

**Comment by samrocketman** [comment]
I don't think @ice1000 implied jd-gui shouldn't support internationalization.  The point he made is the primary language of the developers on this project (at least former) primarily spoke in English.  When filing bug reports to JD-GUI it makes sense to stick to the primary language of development.

**Comment by seamustuohy** [comment]
As someone who came here to search for a related issue I'm glad they asked it. Even if they don't have solid English skills they contributed an issue that is relevant to anyone working with apps in affected character sets. Especially since it seems that Google translate did a *pretty decent* job translating the issue into understandable English. All that to say that I would not discourage people from attempting to contribute issues, etc. even if they have not learned English.  

Now, to my +1. This issue also affects me. When copying Chinese text from source code shown in the GUI and pasting it into any other program it pastes as unrecognizable chars. When I save the sources and open them with my text editor the characters are copied and pasted correctly. 

For example `取消` when copied and pasted from the interface shows up as `������`.

**Comment by donizyo** [comment]
it must be related to clipboard library then :P

**Comment by 1987539447** [comment]
so, how to fix???

**Comment by liach** [comment]
nope, jd gui is dead

**Comment by trigchen** [comment]
A possible work around might be:  
before you copy Chinese text on the text editor GUI, turn on the Chinese input method, then paste the text into target text editor. The Chinese characters may show correctly. 
@seamustuohy

**Comment by starskys** [comment]
> A possible work around might be:
> before you copy Chinese text on the text editor GUI, turn on the Chinese input method, then paste the text into target text editor. The Chinese characters may show correctly.
> @seamustuohy

seems not work

**Comment by ugksoft** [comment]
哎,显示正常, 复制出来,全是空心方块, 在记事本里,显示的是黑方块, 这明摆着坑爹么

**Comment by ArtoriaZhang** [comment]
> I don't think it good to use Chinese on GitHub, especially when you're issuing a non-Chinese project.

so,what happended to ur avatar

**Comment by ArtoriaZhang** [comment]
> when copied and pasted from the interface shows up as

awesome

**Comment by ArtoriaZhang** [comment]
> As someone who came here to search for a related issue I'm glad they asked it. Even if they don't have solid English skills they contributed an issue that is relevant to anyone working with apps in affected character sets. Especially since it seems that Google translate did a _pretty decent_ job translating the issue into understandable English. All that to say that I would not discourage people from attempting to contribute issues, etc. even if they have not learned English.
> 
> Now, to my +1. This issue also affects me. When copying Chinese text from source code shown in the GUI and pasting it into any other program it pastes as unrecognizable chars. When I save the sources and open them with my text editor the characters are copied and pasted correctly.
> 
> For example `取消` when copied and pasted from the interface shows up as `������`.

It works for me. Thanks. @seamustuohy

**Comment by tomuGo** [comment]
这个问题很好搞吧，太不人性化了

**Comment by Yun-Shan** [comment]
> 这个问题很好搞吧，太不人性化了

我刚想回复上面的人才发现这是远古issue挖坟。我之前提了一个临时解决方案的issue，你可以尝试手动替换类文件解决 #268

**Comment by hcgdyx** [comment]
for windows10 PowerShell  run `java -D'file.encoding'=utf-8 -jar .\jd-gui-1.6.6.jar`

**Comment by baishier** [comment]
我用曲线救国的方法解决了这个问题。一两句说不清楚，我写了博客
I solved the problem indirectly.more info about this issues in my blogs
https://www.cnblogs.com/baishier/p/14137326.html

**Comment by liudonghua123** [comment]
Maybe you could use the methods I mentioned in https://github.com/java-decompiler/jd-gui/issues/279#issuecomment-1062524829.
